parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -551,7 +551,15 @@ ensureGPUDrivers() {
     fi
 
     if [[ "${CONFIG_GPU_DRIVER_IF_NEEDED}" = true ]]; then
-        logs_to_events "AKS.CSE.ensureGPUDrivers.configGPUDrivers" configGPUDrivers
+        existing_version="$(nvidia-smi | grep "Driver Version" | cut -d' ' -f3)"
+
+        if [[ "*$existing_version*" == $NVIDIA_DRIVER_IMAGE_TAG ]]; then
+            echo "Existing version matches, validating..."
+            logs_to_events "AKS.CSE.ensureGPUDrivers.validateGPUDrivers" validateGPUDrivers
+        else
+            echo "Existing version $existing_version does not match $NVIDIA_DRIVER_IMAGE_TAG"
+            logs_to_events "AKS.CSE.ensureGPUDrivers.configGPUDrivers" configGPUDrivers
+        fi
     else
```

Comment:
I think we can simplify the whole block:

if CONFIG_GPU_DRIVER_IF_NEEDED; then
  if current_version != desired version; then
    install
  fi
fi
validateGPUDrivers